platform/core/crates/update_notification/tests/lib.rs

```diff
@@ -0,0 +1,155 @@
+use std::sync::{Arc, Mutex};
+use update_notification::*;
+
+struct MockStore {
+    last_notification: Arc<Mutex<Option<LastNotification>>>,
+}
+
+impl MockStore {
+    fn new() -> MockStore {
+        MockStore {
+            last_notification: Arc::new(Mutex::from(None)),
+        }
+    }
+}
+
+impl UpdateNotificationStore for MockStore {
+    fn get_last_notification(&self) -> Option<LastNotification> {
+        self.last_notification.lock().unwrap().clone()
+    }
+
+    fn set_last_notification(&self, last_notification: LastNotification) {
```

Comment:
I'd expect this to be `&self mut`, but it can´t be, because we wrap this in an `Arc` in `Core`, which isn´t mutable. As a workaround for the test, `MockStore` stores `last_notification` in a mutex. Not a huge fan of this, but just copied the approach from `user_session`.

Reply:
I guess you are referring to the fact that Preferences is an immutable object in Core, it could be updated to be mutable if you consider it absolutely needed. Right now Preferences is immutable, since it does not store any state, everything is stored on disk.

Reply:
No super strong opinion on this one. Brings me back to const correctness in C++ :D IMHO, even if the object uses something else (that isn't const correct, like the OS file system API) to hold its logical state, I'd make this logically const correct, so that it's clear that calling a const/non-mut function never has any side effects. But I can also see it the other way.